### PR TITLE
Update dependencies and GitHub actions

### PR DESCRIPTION
**Pull request by rolljee**
## What does this PR do ?

Update deps, remove peer deps

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :

### CODE REVIEW

File: .github/workflows/pull_request.workflow.yml [hunk 1]
```diff
@@ -20,19 +19,18 @@ jobs:
             ${{ runner.os }}-build-${{ env.cache-name }}-
             ${{ runner.os }}-build-
             ${{ runner.os }}-
-      - uses: actions/setup-node@v1.4.4
+      - uses: actions/setup-node@v4
         with:
-          node-version: "14"
+          node-version: "18.17.1"
```
Comment (Kuruyia) [inline .github/workflows/pull_request.workflow.yml:24]: Is it not enough to just use the latest minor of Node.js 18?
```suggestion
          node-version: "18"
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "kuzzle-plugin-prometheus",
-  "version": "4.2.0",
+  "version": "4.1.1",
```
Comment (Kuruyia) [inline package.json]: Why did this rollback from `4.2.0` to `4.1.1`?

File: tsconfig.json [hunk 3]
```diff
@@ -7,11 +7,12 @@
     "moduleResolution": "node",
     "sourceMap": true,
     "baseUrl": ".",
+    "skipDefaultLibCheck": true,
     "resolveJsonModule": true,
     "esModuleInterop": true,
     "experimentalDecorators": true,
     "allowJs": false,
-    "skipLibCheck": true
+    "skipLibCheck": true,
```
Comment (alexandrebouthinon) [inline tsconfig.json]: This is a json file, the trailing coma will not work

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (rolljee) [reply package.json]: Oops this is my bad

File: .github/workflows/pull_request.workflow.yml (hunk 1 above)
Comment (rolljee) [reply .github/workflows/pull_request.workflow.yml:24]: I tried, it seems that mocha, typescript and ESM module are bugged with the last version of Node 18, but it is working in the targeted 18.17.1 idk why ...

**Review by Kuruyia — APPROVED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Pull request merged**